NGitLab.Tests/Docker/GitLabDockerContainer.cs

```diff
@@ -278,127 +275,94 @@ private async Task GenerateCredentialsAsync()
-                          return true;
-                      },
-                  })
-                  .WithDefaultCookies()
-                  .WithLocaleBasedEncoding();
-                using var context = BrowsingContext.New(conf);
-
-                // Change password
-                var result = await context.OpenAsync(GitLabUrl.AbsoluteUri).ConfigureAwait(false);
-                result = await ReloadIfError(context, result);
-
-                TestContext.Progress.WriteLine("Navigating to " + result.Location);
-                if (result.Location.PathName == "/users/password/edit")
+                // Use Playwright to launch Chromium
+                using var playwright = await Playwright.CreateAsync();
+                await using var browser = await playwright.Chromium.LaunchAsync(new BrowserTypeLaunchOptions
                 {
-                    TestContext.Progress.WriteLine("Creating root password");
-                    var form = result.Forms["new_user"];
-                    if (form == null)
-                        throw new InvalidOperationException("Cannot set the root password. The page doesn't contain the form 'new_user'");
-
-                    ((IHtmlInputElement)form["user[password]"]).Value = AdminPassword;
-                    ((IHtmlInputElement)form["user[password_confirmation]"]).Value = AdminPassword;
-                    result = await form.SubmitAsync();
-                    TestContext.Progress.WriteLine("Navigating to " + result.Location);
-                }
+                    // Headless = false,   // Uncomment to have browser window visible
+                    // SlowMo = 1000,      // Slows down Playwright operations by the specified amount of ms.
+                });
+
+                await using var browserContext = await browser.NewContextAsync();
+
+                var page = await browserContext.NewPageAsync();
+                await page.GotoAsync(GitLabUrl.AbsoluteUri);
+
+                TestContext.Progress.WriteLine("Generating Credentials");
+
+                var url = await GetCurrentUrl(page);
 
                 // Login
-                if (result.Location.PathName == "/users/sign_in")
+                if (url == "/users/sign_in")
                 {
-                    result = await SignIn(context);
+                    await page.Locator("form#new_user input[name='user[login]']").FillAsync(AdminUserName);
+                    await page.Locator("form#new_user input[name='user[password]']").FillAsync(AdminPassword);
+
+                    var checkbox = page.Locator("form#new_user input[type=checkbox][name='user[remember_me]']");
+                    await checkbox.CheckAsync(new LocatorCheckOptions { Force = true });
+
+                    await page.RunAndWaitForResponseAsync(async () =>
+                    {
+                        await page.EvalOnSelectorAsync("form#new_user", "form => form.submit()");
```

Comment:
Could it simply be replaced with a click on the form submit button?

Reply:
Clicking on a button means more maintenance. If the UI change => KO.